### PR TITLE
Prefer more specific path map key

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/roslyn/issues/46061

### CODE REVIEW

**Comment by jcouv** [comment]
Do we need to document this as a breaking change in some way?

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Comment by tmat** [comment]
> Do we need to document this as a breaking change in some way?

Maybe. I wouldn't think this would cause a serious issue in practice, but if we want to track it in some breaking changes list I can add a note there. LMK.

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs [hunk 1]
```diff
@@ -1482,6 +1482,11 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string? ba
                                                  new CSharpRequiredLanguageVersion(MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion())), Location.None));
             }
 
+            // Sort so that more specific keys precede less specific.
+            // When mapping a path we find the first key in the array that is a prefix of the path.
+            // If multiple keys are prefixes of the path we want to use the longest (more specific) one for the mapping.
+            pathMap = pathMap.Sort((x, y) => -x.Key.Length.CompareTo(y.Key.Length));
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs:1488]: Why are we only doing this for C#? VB also supports path mapping and believe we'd want to be consistent here.

Comment (tmat) [reply src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs:1488]: Oh, good catch. Forgot about VB. Will add this to VB as well.

**Review by jaredpar — APPROVED** [review]

**Review by RikkiGibson — APPROVED** [review]

**Pull request merged**